xdsl/irdl/irdl.py

```diff
@@ -429,6 +429,115 @@ def infer(self, constraint_vars: dict[str, Attribute]) -> Attribute:
         return self.constr.infer(constraint_vars)
 
 
+class RangeConstraint(ABC):
+
+    @abstractmethod
+    def verify(
+        self, attrs: Sequence[Attribute | None], constraint_vars: dict[str, Attribute]
+    ) -> None:
+        """
+        Check if the range satisfies the constraint,
+        or raise an exception otherwise.
+        The range can contain Nones, which represent an attribute not to be checked.
```

Comment:
I forgot if you told me before, but why is there possibly attributes that shouldn't be checked?

Reply:
I'm honestly not sure this is needed; will check if it's an issue to revert this for now!

Idea was to use for inference of not yet fully known sequences

Reply:
Done, forgetting about None for now, it was only needed for Pyright at this point.
I suspect with more complex RangeConstraint, it might becone useful, but let's think about it then